app/javascript/find/controllers/filter_search_controller.js

```diff
@@ -1,58 +1,38 @@
 import { Controller } from '@hotwired/stimulus'
+import Mustache from 'mustache'
 
 // Connects to data-controller="filter-search"
 export default class extends Controller {
-  static targets = ['optionsList', 'searchInput', 'legend']
+  static targets = ['optionsList', 'searchInputContainer', 'legend']
 
   static instanceCounter = 0
 
   connect () {
     const legend = this.element.querySelector('legend')
     legend.dataset.filterSearchTarget = 'legend'
 
-    // This will be unique for each filter-search instance
+    // Assign a unique ID for input/label association
     this.instanceId = this.constructor.instanceCounter++
 
-    const searchInput = this.createSearchInput()
-    this.optionsListTarget.before(searchInput)
-  }
+    const template = document.getElementById('filter-search-input-template').innerHTML
+    const html = Mustache.render(template, {
+      id: `${this.identifier}-${this.instanceId}-input`,
+      label: this.legendTarget.innerText
+    })
 
-  createSearchInput () {
-    const container = document.createElement('div')
-    container.classList.add('filter-search__search')
-
-    const inputId = `${this.identifier}-${this.instanceId}-input`
-    const labelText = `${this.legendTarget.innerText}`
-
-    container.innerHTML = `
-      <label for="${inputId}" class="govuk-label govuk-visually-hidden">
-        ${labelText}
-      </label>
-      <input type="search" id="${inputId}"
-        class="govuk-input govuk-!-margin-bottom-1"
-        autocomplete="off"
-        placeholder="Search"
-        data-action="input->${this.identifier}#search"
-        data-filter-search-target="searchInput">
-    `
-
-    return container
+    this.searchInputContainerTarget.innerHTML = html
   }
 
   search () {
     const optionItems = this.optionsListTarget.children
-    const searchValue = this.searchInputTarget.value.toLowerCase()
+    const searchValue = this.searchInputContainerTarget.querySelector('input')?.value?.toLowerCase() || ''
 
     this.toggleItems(optionItems, searchValue)
   }
 
   toggleItems (items, searchValue) {
-    Array.from(items).forEach(function (item) {
-      if (item.textContent.toLowerCase().indexOf(searchValue) > -1) {
-        item.style.display = ''
-      } else {
-        item.style.display = 'none'
-      }
+    Array.from(items).forEach(item => {
+      item.style.display = item.textContent.toLowerCase().includes(searchValue) ? '' : 'none'
```

Comment:
Last thing, by adding the `searchInputTarget` back, these changes shouldn't be necessary again, right?